### ISSUE TITLE
[FEA] Decimal 128 Support: Audit/Cleanup

### ISSUE THREAD

**Issue by revans2** [NVIDIA/spark-rapids#3890]
As a part of adding in support for Decimal 128. We did a most of it one operator at a time. Once this is merged into the main branch again. 

- [x] We should go back through and verify that we didn't miss any operators, that all of the `TypeSig` helpers like `TypeSig.gpuNumeric` are either replaced with `numeric` or at least refer to `DECIMAL_128_FULL`.
- [x] Might be good to rename `DECIMAL_128_FULL` to just `DECIMAL_128`. 
- [x] We should go through the tests and make sure that the data gens are all updated to have decimal support correctly in them.
- [x] We should also go through all of the documentation and make sure it has been properly updated to refer to what we can support now.
Just generally go through and clean up.

**Comment by kuhushukla** [comment]
I have been going through the execs and exprs and will address other issues mentioned above shortly.

**Comment by kuhushukla** [comment]
@sameerz I reopened this since we can use this to get even a couple more changes in when hash partitioning is in and rework that I suggested above. Please feel free to close and open just a smaller task if you would prefer.

**Comment by nartal1** [comment]
We are testing decimal_128 in join_tests by including  `decimal_128_gens` in `all_gen` in this PR https://github.com/NVIDIA/spark-rapids/pull/4470 and order by tests are testing decimal_128. Closing.
